### PR TITLE
Use kibana server hostname for puppeteer to access pages

### PR DESCRIPTION
**Pull request by zhongnansu**
*Issue #, if available:*
Currently the url for puppeteer to access is a complete url passed from UI, including the schema, hostname and port. However this brings some issues when deploying on a cloud service such as AWS ELB, which has settings for security group. Since puppeteer is only accessing the url from localhost, it doesn't even need to go the ELB first and route back to itself. Instead it can access `http://localhost:5601:/app/dashboard#/<dashboard_id>` directly.

*Description of changes:*
- This is a temporary fix to replace url prefix with a hardcoded `LOCAL_HOST = "http://localhost:5601"` constant. However since the hostname and port number are configured in `kibana.yml` file, which the new Kibana plugin platform doesn't provide access now. This issue is documented #98  and tracked once there is a proper solution.
- remove unused dependencies, move some packages from `dependency` to `devDependency`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: kibana-reports/server/routes/utils/constants.ts [hunk 1]
```diff
@@ -66,3 +66,5 @@ export enum TRIGGER_TYPE {
 }
 // https://www.elastic.co/guide/en/elasticsearch/reference/6.8/search-request-from-size.html
 export const DEFAULT_MAX_SIZE = 10000;
+
+export const LOCAL_HOST = 'http://localhost:5601';
```
Comment (davidcui-amzn) [inline kibana-reports/server/routes/utils/constants.ts:70]: Should we be hard-coding the host like this?

File: kibana-reports/server/routes/utils/dataReportHelpers.ts [hunk 2]
```diff
@@ -1,4 +1,3 @@
-import { async } from 'rxjs/internal/scheduler/async';
```
Comment (davidcui-amzn) [inline kibana-reports/server/routes/utils/dataReportHelpers.ts:1]: Are the changes in this file related to the title of the PR? I feel this is a different optimization refactor

Comment (zhongnansu) [reply kibana-reports/server/routes/utils/dataReportHelpers.ts:1]: That's just an unused import

File: kibana-reports/server/routes/utils/constants.ts (hunk 1 above)
Comment (zhongnansu) [reply kibana-reports/server/routes/utils/constants.ts:70]: Ideally, we should extract the server host/port from `kibana.yml`, but the new platform applies some restrictions to access certain fields. More details can be found in this issue #98

File: kibana-reports/server/routes/utils/dataReportHelpers.ts (hunk 2 above)
Comment (davidcui-amzn) [reply kibana-reports/server/routes/utils/dataReportHelpers.ts:1]: Not just this line, I'm referring to the other changes in the file as well- meant to highlight the whole file

File: kibana-reports/server/routes/utils/constants.ts (hunk 1 above)
Comment (davidcui-amzn) [reply kibana-reports/server/routes/utils/constants.ts:70]: Can we get the local host from a `window.location` or `document` method instead? I think those should be fine to access in the new platform.

Comment (zhongnansu) [reply kibana-reports/server/routes/utils/constants.ts:70]: That `window.location` won't work on server side. The UI side still keeps the usage of `window.location` which is fine.

Comment (davidcui-amzn) [reply kibana-reports/server/routes/utils/constants.ts:70]: Ack, miss from me- forgot this was on server-side

**Review by davidcui-amzn — APPROVED** [review]
LGTM- just some small comments and questions to address

File: kibana-reports/server/routes/utils/dataReportHelpers.ts (hunk 2 above)
Comment (zhongnansu) [reply kibana-reports/server/routes/utils/dataReportHelpers.ts:1]: The other change is just rewriting the import to typescript style.

**Review by akbhatta — APPROVED** [review]

**Pull request merged**